Common/cpp/Tools/BackgroundQueue.cpp

```diff
@@ -0,0 +1,97 @@
+#include "BackgroundQueue.h"
+
+#include <utility>
+
+namespace reanimated {
+
+BackgroundQueue::BackgroundQueue(const std::string &name) : name_(name) {
+  thread_ = std::thread([this] {
+#if __APPLE__
+    pthread_setname_np(name_.c_str());
+#endif
+    runLoop();
+  });
+#ifdef ANDROID
+  pthread_setname_np(thread_.native_handle(), name_.c_str());
+#endif
+}
+
+BackgroundQueue::~BackgroundQueue() {
+  running_ = false;
+  cv_.notify_all();
+  thread_.join();
+}
+
+jsi::Value BackgroundQueue::get(
+    jsi::Runtime &rt,
+    const jsi::PropNameID &propName) {
+  auto name = propName.utf8(rt);
+  if (name == "toString") {
+    return jsi::Function::createFromHostFunction(
+        rt,
+        propName,
+        0,
+        [this](jsi::Runtime &rt, const jsi::Value &, const jsi::Value *, size_t)
+            -> jsi::Value {
+          return jsi::String::createFromUtf8(rt, toString());
+        });
+  }
+  if (name == "name") {
+    return jsi::String::createFromUtf8(rt, name_);
+  }
+  return jsi::Value::undefined();
+}
+
+std::vector<jsi::PropNameID> BackgroundQueue::getPropertyNames(
+    jsi::Runtime &rt) {
+  std::vector<jsi::PropNameID> result;
+  result.push_back(jsi::PropNameID::forUtf8(rt, "toString"));
+  result.push_back(jsi::PropNameID::forUtf8(rt, "name"));
```

Comment:
How about moving it to initialization list here?

Reply:
I've tried it once but it didn't work for me. 🤷 Also, other libs usually cache these props but in this case it sounds like an overkill to me.

Reply:
🤔